spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/ConditionalOnConfigMapPropertiesRetryEnabled.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2013-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.commons.config;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.cloud.kubernetes.commons.ConditionalOnKubernetesConfigEnabled;
+import org.springframework.cloud.kubernetes.commons.ConditionalOnKubernetesEnabled;
+
+/**
+ * {@link org.springframework.context.annotation.Conditional @Conditional} that only
+ * matches when Spring Cloud Kubernetes, Kubernetes ConfigMap property sources and
+ * Kubernetes ConfigMap property sources fail fast (thus retry) are enabled.
+ *
+ * @author Isik Erhan
+ */
+@Target({ ElementType.TYPE, ElementType.METHOD })
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@Inherited
+@ConditionalOnKubernetesEnabled
+@ConditionalOnKubernetesConfigEnabled
+@ConditionalOnProperty(prefix = ConfigMapConfigProperties.PREFIX, name = "fail-fast", havingValue = "true")
```

Comment:
what if we extract `@ConditionalOnProperty(prefix = ConfigMapConfigProperties.PREFIX, name = "fail-fast", havingValue = "true")` to its own annotation called `ConditionalOnKubernetesConfigRetryEnabled` and use that here: 

```
@ConditionalOnKubernetesEnabled
@ConditionalOnKubernetesConfigEnabled
@ConditionalOnKubernetesConfigRetryEnabled
```

reads pretty awesome, to me. what do you think?

Reply:
Actually, the names `@ConditionalOnConfigMapPropertiesRetryEnabled` and `@ConditionalOnKubernetesConfigRetryEnabled` both refer to the same logical condition. However, I like the name you suggested; it's clearer and a bit shorter. Maybe we can rename `@ConditionalOnConfigMapPropertiesRetryEnabled` itself to `@ConditionalOnKubernetesConfigRetryEnabled` and extract `@ConditionalOnProperty(prefix = ConfigMapConfigProperties.PREFIX, name = "fail-fast", havingValue = "true")` to something like `@ConditionalOnKubernetesConfigFailFastEnabled`. It would look like the following:

ConditionalOnKubernetesConfigRetryEnabled .java:
```java
@ConditionalOnKubernetesEnabled
@ConditionalOnKubernetesConfigEnabled
@ConditionalOnKubernetesConfigFailFastEnabled
public @interface ConditionalOnKubernetesConfigRetryEnabled {
...
```

ConditionalOnKubernetesConfigFailFastEnabled.java
```java
@ConditionalOnProperty(prefix = ConfigMapConfigProperties.PREFIX, name = "fail-fast", havingValue = "true")
public @interface ConditionalOnKubernetesConfigFailFastEnabled {
...
```

The only thing that I don't like with this approach is the extracted condition `@ConditionalOnKubernetesConfigFailFastEnabled` will not check if Kubernetes itself or Kubernetes config is enabled or not. This could make it a bit useless when someone wants it to use in another place since it does not actually check if "Kubernetes config fail-fast" feature is enabled, it only checks if the property `*.fail-fast` is equal to `true`.

Reply:
```
@ConditionalOnKubernetesEnabled
@ConditionalOnKubernetesConfigEnabled
@ConditionalOnKubernetesConfigFailFastEnabled
public @interface ConditionalOnKubernetesConfigRetryEnabled {
```

this! very much this; good name, I like it more than what I proposed initially. 

As to you concern, imo : if we document the fact that this does not check neither for kubernetes nor config map - it's _documented_ now, so no problem? :)